src/org/omegat/core/team2/impl/GITRemoteRepository2.java

```diff
@@ -382,6 +382,39 @@ public static boolean deleteDirectory(File path) {
         return (path.delete());
     }
 
+    /**
+     * Retrieve default branch name from repository.
+     * @param repository target repository.
+     * @param repositoryUrl Remote repository URL
+     * @return default branch name, ordinary "main" (recent popular) or "master" (old default)
+     */
+    public static String getDefaultBranchName(final Repository repository, final String repositoryUrl) {
```

Comment:
Is the repository URL not available directly from the `Repository` object? If so we could simplify this a bit.

Reply:
It looks like you can do

```
repository.getConfig().getString("remote", "origin", "url");
```
https://stackoverflow.com/a/38062680/448068

Reply:
I wonder in what cases would `repository.getBranch()` return null though.

`getBranch` calls `getFullBranch` which says

```
	 * @return name of current branch (for example {@code refs/heads/master}),
	 *         an ObjectId in hex format if the current branch is detached, or
	 *         {@code null} if the repository is corrupt and has no HEAD
	 *         reference.
```

So do we have to worry about the null case? If we have a corrupt repository it seems like we have bigger problems.

Reply:
Just because it is nullable
https://github.com/eclipse/jgit/blob/88e924e86beaa7e4911f4496e847a10833ac1c15/org.eclipse.jgit/src/org/eclipse/jgit/lib/Repository.java#L1000-L1015

```
	@Nullable
	public String getFullBranch() throws IOException {
		Ref head = exactRef(Constants.HEAD);
		if (head == null) {
			return null;
		}
		if (head.isSymbolic()) {
			return head.getTarget().getName();
		}
		ObjectId objectId = head.getObjectId();
		if (objectId != null) {
			return objectId.name();
		}
		return null;
	}
```

Reply:
And detached HEAD can be null.
When `git checkout <hash>`

Reply:
It says `an ObjectId in hex format if the current branch is detached`.

Reply:
> And detached HEAD can be null.

No, it is my misunderstanding. I don't know a reason other than corrupted repository.